platform/linux-generic/pktio/socket_mmap.c

```diff
@@ -243,7 +247,21 @@ static inline unsigned pkt_mmap_v2_rx(pktio_entry_t *pktio_entry,
 			memmove(mac, mac + vlan_len, 2 * _ODP_ETHADDR_LEN);
 			type  = (uint16_t *)(uintptr_t)
 				(mac + 2 * _ODP_ETHADDR_LEN);
+
+			#ifdef TP_STATUS_VLAN_TPID_VALID
 			*type = odp_cpu_to_be_16(tp_hdr->tp_vlan_tpid);
+			#else
+			/* Fallback for old kernels (< v3.14) */
+			uint16_t *type2;
+
+			type2 = (uint16_t *)(uintptr_t)(mac + (2 * _ODP_ETHADDR_LEN) + vlan_len);
```

Comment:
What is the point of the intermediate cast to uintptr_t?

Btw, this line assumes mac is correctly aligned for the alignment requirement of uint16_t and the uintptr_t does not change that. But it is not a new assumption so it should be ok here too.

Reply:
Some clang versions in the CI are trigger happy to complain about casts, so used the same patter as couple lines above.